Bump to K8s 1.35

skitt opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Chores**
  * Updated Go runtime to 1.25.0 and upgraded Kubernetes APIs plus numerous transitive dependencies for compatibility and tooling updates.
* **Improvements**
  * Improved list/watch semantics used by sync controllers and related tests to enhance reliability of namespace and resource synchronization.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

skitt committed:
Bump to K8s 1.35

Signed-off-by: Stephen Kitt <skitt@redhat.com>

tpantelis commented:
The `WatchAndSignalOnChange` unit tests are failing (even after adding a 500ms delay in the tests) but, strangely, they succeed locally for me (I ran it hundreds of times).  @skitt do they pass locally for you?

skitt commented:
> The `WatchAndSignalOnChange` unit tests are failing (even after adding a 500ms delay in the tests) but, strangely, they succeed locally for me (I ran it hundreds of times). @skitt do they pass locally for you?

They fail locally for me…

tpantelis commented:
The unit tests now fail locally for me and I tracked down the issue. It's due to a beta feature named `WatchListClient`:

_The Kubernetes WatchList Client feature enables clients to retrieve an initial state of resources and then receive continuous updates via a single, efficient stream of "watch" events, significantly reducing memory consumption and improving API server stability compared to traditional separate list and watch operations._.

Basically the fake K8s clients haven't yet been updated to support this new feature. Apparently this affected many unit tests in the field so they introduced a mechanism to workaround it, defined [here](https://github.com/kubernetes/client-go/blob/ff70f47ee286a98dfb5c01c2f8a9a8661b6c7470/tools/cache/listwatch.go#L134).

I'll push a commit to address it. The resource syncer tests are also affected.

tpantelis committed:
Use cache.ToListWatcherWithWatchListSemantics with informers

...as a workaround due to the fake K8s clients not yet supporting
the WatchListClient beta feature.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis commented on pkg/configmap/config_map_test.go:
```diff
@@ -118,6 +118,11 @@ func testWatchAndSignalOnChange() {
+		// Even though we received the watch action, there's still a small window where the processing and
+		// notifying of initial resources by the informer may not have completed yet, and thus we don't want to
+		// create a ConfigMap yet, so sleep a bit.
+		time.Sleep(500 * time.Millisecond)
```
This has been removed, please resolve the comment.

tpantelis commented on pkg/configmap/config_map_test.go:
```diff
@@ -134,7 +139,7 @@ func testWatchAndSignalOnChange() {
 			createConfigMap(configMap1, nil)
 
-			Eventually(signalCh).Should(Receive(Equal(signalNum)))
+			Eventually(signalCh).Within(2 * time.Second).Should(Receive(Equal(signalNum)))
```
This isn't an issue, please resolve the comment.

skitt commented:
Excellent, thanks @tpantelis!
